File: src/main/java/x/samantha/commandsjda/abs/Module.java
package x.samantha.commandsjda.abs;

import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

public class Module {

    String name;
    Boolean isEnabled;
    List<ICommand> commands = new ArrayList<>();

    public Module(String name) {
        this.name = name;
    }

    Module addCommand(ICommand command) {
        this.commands.add(command);
        return this;
    }

    Module addCommands(ICommand... commands) {
        this.commands.addAll(Arrays.asList(commands));
        return this;
    }

    Module setEnabled(Boolean enabled) {
        this.isEnabled = enabled;
        return this;
    }

    public String getName() {
        return name;
    }

    public Boolean isEnabled() {
        return isEnabled;
    }

    public List<ICommand> getCommands() {
        return commands;
    }
}

File: src/main/kotlin/x/samantha/commandsjda/abs/Module.kt
package x.samantha.commandsjda.abs

import java.util.ArrayList

class Module(val name: String) {
    var commands: MutableList<ICommand> = ArrayList()
    private var isEnabled: Boolean = true

    /**
     * Add a command to the module
     * @param command
     * The command to be added to the module
     */
    fun addCommand(command: ICommand): Module {
        this.commands.add(command)
        return this
    }

    /**
     * Add commands to the module
     * @param commands
     * The commands to be added to the module
     */
    fun addCommands(vararg commands: ICommand): Module {
        commands.forEach { this.commands.add(it) }
        return this
    }

    /**
     * Enable or disable this module
     * @param value
     * Whether to enable or disable the module
     */
    fun setEnabled(value: Boolean) {
        this.isEnabled = value
    }

    /**
     * Whether the module is enabled
     * @return Boolean
     */
    fun getEnabled(): Boolean {
        return isEnabled
    }
}
